main.rb

```diff
@@ -103,127 +224,123 @@ def update_project(project, file_response, access_token)
+        max_failure_percentage = calc_percent($ac_max_failure_percentage, 100)
+
+        if max_failure_percentage <= failure_percentage || !result_summary[:ERROR].nil?
+          puts "The number of failures in the plan exceeded the maximum rate. The process is being stopped. #{data[:test_result_status_counts]}"
+          exit(1)
+        else
+          puts("Number of failures is below the maximum rate. Process continues. #{data[:test_result_status_counts]}")
+        end
+      else
+        warn_message = "To calculate the failure rate, the following values must be greater than 0:" \
+          "\nAC_TESTINIUM_MAX_FAIL_PERCENTAGE: #{$ac_max_failure_percentage}" \
+          "\nTestinium Result Failure Summary: #{result_failure_summary}"
+        puts warn_message
+      end
+      return
+    elsif (res.kind_of? Net::HTTPClientError)
+      puts(get_parsed_response(res.body)[:message])
+      count += 1
     else
-      puts("Number of failures is below the maximum rate. Process continues. #{data[:test_result_status_counts]}")
+      puts("Error while starting Plan. Response from server: #{get_parsed_response(res.body)}")
+      count += 1
     end
-  else
-    warn_message = "To calculate the failure rate, the following values must be greater than 0:"\
-    "\nAC_TESTINIUM_MAX_FAIL_PERCENTAGE: #{$ac_max_failure_percentage}"\
-    "\nTestinium Result Failure Summary: #{result_failure_summary}"
-    puts warn_message
-  end
-end
-
-def check_status(test_timeout, access_token)
-  if test_timeout <= 0
-    puts('Plan timed out')
-    exit(1)
-  end
-  uri = URI.parse("https://testinium.io/Testinium.RestApi/api/plans/#{$plan_id}/checkIsRunning")
-  req = Net::HTTP::Get.new(uri.request_uri,
-                           { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{access_token}", 'current-company-id' => "#{$company_id}"})
-  res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
-    http.request(req)
-  end
-  case res
-  when Net::HTTPClientError, Net::HTTPServerError
-    puts "\nError checking status: #{res.code} (#{res.body})\n\n"
-    exit(1)
-  end
-  is_running = JSON.parse(res.body, symbolize_names: true)[:running]
-  if is_running
-    puts('Plan is still running...')
-    sleep($time_period)
-    check_status(test_timeout - $time_period, access_token)
-  else
-    puts('Execution finished')
-    true
   end
+  exit(1)
 end
 
 access_token = login()
-raise("Cannot login to Testinium with given credentials") if access_token.nil?
-
+check_status(access_token)
 project = find_project(access_token)
-raise("Cannot find projects in Testinium") if project.nil?
-
-puts "Uploading #{$file_name} to Testinium..."
 file_response = upload(access_token)
-raise("File cannot upload to Testinium successfully") if file_response.nil?
-
-puts "Testinium project is updating..."
 update_status = update_project(project, file_response, access_token)
```

Comment:
With the latest changes in here, we can remove the `update_status` variable. (useless assignment)